### PR TITLE
[tx] Introduce LogitsProcessor  and skip full logits computation during prefill

### PR DESCRIPTION
**Pull request by raulchen**
Summary

  - Add skip_prompt_logits parameter to Llama3 and Qwen3 models to avoid materializing the full [B, T, V] logits tensor during prefill
  - Add LogitsProcessor abstraction similar to huggingface transformers / vllm / sglang, to unify the implementation for all models
  - When prompt_logprobs=False (the common case), only compute logits for the last token
  - Add parametrized tests for both models verifying output shape and generation equivalence

  Motivation

  During prefill, only the last token's logits are needed to start decoding. Computing logits for all prompt tokens requires a
[B, T, V] matmul where V (vocab size) is typically 32K-128K. This is wasteful when prompt_logprobs is not requested.

  This optimization benefits:
  - Standard inference/chat (most common)
  - RL training (only generation logprobs needed, not prompt logprobs)

  Test plan

  - test_last_token_logits_only[llama3] - verifies output shape and generation equivalence
  - test_last_token_logits_only[qwen3] - verifies output shape and generation equivalence
  - Existing generator tests pass

Benchmark results https://github.com/NovaSky-AI/SkyRL/issues/891#issuecomment-3775792201

### CODE REVIEW

**Comment by pcmoritz** [comment]
I'm currently trying to find a simple setting where this avoids OOM (a real end-to-end workload), if you already have one let me know :)

**Comment by raulchen** [comment]
@pcmoritz I'm compiling my benchmark results from this script https://github.com/NovaSky-AI/SkyRL/pull/892
Will update them to the main issue soon.

**Comment by pcmoritz** [comment]
End-to-end example that works with this PR, but not master:
```
uv run --extra gpu --extra tinker -m tx.tinker.api     --base-model Qwen/Qwen3-8B     --backend-config '{"max_lora_adapters": 3, "max_lora_rank": 1, "tensor_parallel_size": 8, "train_micro_batch_size": 2, "sample_max_num_sequences": 776}' > out.log

uv run --with wandb --with tinker==0.3.0 rl_loop.py     base_url=http://localhost:8000     model_name="Qwen/Qwen3-8B"     lora_rank=1 max_length=3072 max_tokens=3072
```
